inginious/frontend/plugins/UNCode/static/js/task_files_tab.js

```diff
@@ -33,6 +38,20 @@ jQuery(document).ready(function () {
         });
     };
 
+    /**
+     * Patch studio_task_file_delete to detect when a root file related to test is removed and remove that test too.
+     * Remove test is only applied if the environment is multiLang or data science
+     */
+    _this.studio_task_file_delete = (path) => {
+        if (!confirm("Are you sure you want to delete this?") || !studio_task_file_delete_tab(path))
+            return;
+        studio_update_file_tabs({"action": "delete", "path": path});
+        if ((isMultiLang || isDataScience) && isRoot(path)) {
+            const filename = path.substr(1);
+            removeTestByFilename(filename);
+        }
+    };
```

Comment:
I think we should also try to remove the file in the `public/` folder in case it is located there. It might generate some trash after some time.